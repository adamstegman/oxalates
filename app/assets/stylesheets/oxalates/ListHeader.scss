@import "./constants";
@import "./lists";
@import "./ActionsMenu.scss";

$title_height: 38px;
$title_padding: 15px 0;
$title_font_family: "MarkerFelt-Thin", $font_family;
$title_font_size: $title_height;

.title {
  padding: $title_padding;
  -webkit-flex-basis: $title_height;
  flex-basis: $title_height;

  background-color: $title_background_color;
  font-family: $title_font_family;
  font-size: $title_font_size;
  text-align: center;
}
