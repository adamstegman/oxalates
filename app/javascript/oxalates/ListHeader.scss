@import "./constants";
@import "./body-form";
@import "./hidden";
@import "./lists";

$title_height: 38px;
$title_padding: 15px 0;
$title_font_family: "MarkerFelt-Thin", $font_family;
$title_font_size: $title_height;

$actions_height: 44px;
$actions_background_color: lighten($title_background_color, 20%);
$actions_font_size: 16px;
$actions_gutter_width: 0.5em;
$session_padding: $actions_gutter_width 0 $actions_gutter_width $actions_gutter_width;
$session_width: 4em;
$search_padding: $actions_gutter_width;
$search_width: 9em;
$add_food_padding: $session_padding;
$add_food_width: 2em;
$edit_food_padding: $session_padding;
$edit_food_width: 2em;

.title {
  padding: $title_padding;
  -webkit-flex-basis: $title_height;
  flex-basis: $title_height;

  background-color: $title_background_color;
  font-family: $title_font_family;
  font-size: $title_font_size;
  text-align: center;
}

.actions {
  display: -webkit-flex;
  -webkit-align-items: center;
  -webkit-justify-content: space-between;
  display: flex;
  align-items: center;
  justify-content: space-between;

  min-height: $actions_height;

  background-color: $actions_background_color;
  font-size: $actions_font_size;
  a {
    text-decoration: none;
  }

  .session {
    margin: $session_padding;
    min-width: $session_width;
    max-width: $session_width;
    &.logout, &.login {
      display: block;
    }

    text-align: center;
  }

  .search {
    min-width: $search_width;
    padding: $search_padding;
  }

  .add-food {
    padding: $add_food_padding;
    min-width: $add_food_width;

    text-align: center;
  }

  .edit-food {
    padding: $edit_food_padding;
    min-width: $edit_food_width;

    text-align: center;
  }
}

.session {
  @extend %body-form;
}

.search {
  label, input[type='submit'] {
    @extend %hidden;
  }
  input[type='search'] {
    font-size: $actions_font_size;
    width: 100%;
  }
}
