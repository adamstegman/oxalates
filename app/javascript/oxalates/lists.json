[
  {
    "id": "all",
    "name": "All",
    "bottom_threshold": null,
    "top_threshold": null
  },
  {
    "id": 1,
    "name": "Very High",
    "bottom_threshold": 50,
    "top_threshold": null
  },
  {
    "id": 2,
    "name": "High",
    "bottom_threshold": 15,
    "top_threshold": 50
  },
  {
    "id": 3,
    "name": "Moderate",
    "bottom_threshold": 5,
    "top_threshold": 15
  },
  {
    "id": 4,
    "name": "Low",
    "bottom_threshold": null,
    "top_threshold": 5
  }
]
