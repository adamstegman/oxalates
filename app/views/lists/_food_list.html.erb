<ul class="food-list">
<% foods.each do |food| %>
  <li class="food-list-item <%= list_class_name(food.list) %>"><%= food.name %></li>
<% end %>
</ul>
