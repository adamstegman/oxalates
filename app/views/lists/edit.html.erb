<%= form_tag list_path(@list.id), method: :put do %>
  <ul class="food-list">
  <% @list.foods.each do |food| %>
    <li>
      <%= link_to food.name, edit_food_path(food), class: "edit-food-list-item #{list_class_name(food.list)}" %>
      <%= hidden_field_tag 'list[foods][][id]', food.id %>
      <%= label_tag 'list[foods][][name]', "Name", class: 'food-list-item-name' %>
      <%= text_field_tag 'list[foods][][name]', food.name, class: 'food-list-item-name' %>
    </li>
  <% end %>
  </ul>
<% end %>
