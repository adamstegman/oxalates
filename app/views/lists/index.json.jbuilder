json.lists(@lists) do |list|
  json.(list, :id, :name, :bottom_threshold, :top_threshold)
end
