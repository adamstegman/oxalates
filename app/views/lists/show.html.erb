<%= render "food_list", foods: @list.foods %>
