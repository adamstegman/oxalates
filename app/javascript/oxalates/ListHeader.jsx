import React from 'react';
import PropTypes from 'prop-types';

import { cssify } from './cssify';
import { listPropType } from './listPropType';
import { ActionsMenu } from './ActionsMenu';

import './ListHeader.scss';

export class ListHeader extends React.Component {
  getListHeader(list) {
    if (list.name === "All") {
      return 'All Foods';
    }

    return `${list.name} Oxalates`;
  }

  render() {
    const {
      list,
    } = this.props;

    const listClassName = cssify(this.props.list.name);
    const header = this.getListHeader(this.props.list);

    return (
      <div>
        <h1 className={`title ${listClassName}`}>{header}</h1>
        <ActionsMenu />
      </div>
    );
  }
};

ListHeader.propTypes = {
  list: listPropType.isRequired,
};
