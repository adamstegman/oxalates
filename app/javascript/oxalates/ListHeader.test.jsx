import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import lists from './__mocks__/lists.json';
import { ListHeader } from './ListHeader';

const mockStore = configureStore();
const state = {
  foodList: {
    query: 'test',
  },
  session: {
    authenticating: false,
    password: '',
  },
};

test('ListHeader renders the All list title', () => {
  const store = mockStore(state);
  const component = renderer.create(
    <Provider store={store}>
      <ListHeader list={lists[0]} />
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('ListHeader renders the Very High list title', () => {
  const store = mockStore(state);
  const component = renderer.create(
    <Provider store={store}>
      <ListHeader list={lists[1]} />
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
