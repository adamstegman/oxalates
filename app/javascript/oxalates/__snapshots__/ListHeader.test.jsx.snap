// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListHeader renders the All list title 1`] = `
<div>
  <h1
    className="title all"
  >
    All Foods
  </h1>
  <div
    className="actions"
  >
    <button
      className="action login"
      onClick={[Function]}
      type="button"
    >
      Log in
    </button>
    <form
      className="search"
      onSubmit={[Function]}
    >
      <label
        htmlFor="search"
      >
        Search
      </label>
      <input
        name="search"
        onChange={[Function]}
        placeholder="Search"
        type="search"
        value="test"
      />
      <input
        type="submit"
        value="Search"
      />
    </form>
  </div>
</div>
`;

exports[`ListHeader renders the Very High list title 1`] = `
<div>
  <h1
    className="title very-high"
  >
    Very High Oxalates
  </h1>
  <div
    className="actions"
  >
    <button
      className="action login"
      onClick={[Function]}
      type="button"
    >
      Log in
    </button>
    <form
      className="search"
      onSubmit={[Function]}
    >
      <label
        htmlFor="search"
      >
        Search
      </label>
      <input
        name="search"
        onChange={[Function]}
        placeholder="Search"
        type="search"
        value="test"
      />
      <input
        type="submit"
        value="Search"
      />
    </form>
  </div>
</div>
`;
